File: examples/eApi v1.0/dotNET/KHGTWDemo/KHGatewayClientExample/Communication/DataObjects/Response/OneclickEchoResponse.cs
using System.Text;
using KHGatewayClientExample.Communication.DataObjects.Ext;

namespace KHGatewayClientExample.Communication.DataObjects.Response;

public class OneclickEchoResponse : SignBase
{
    public string? OrigPayId { get; set; }
    public int ResultCode { get; set; }
    public string? ResultMessage { get; set; }

    public List<Extension>? Extensions { get; set; }

    public override string ToSign()
    {
        var sb = new StringBuilder();
        Add(sb, OrigPayId);
        Add(sb, Dttm);
        Add(sb, ResultCode);
        Add(sb, ResultMessage);
        DeleteLast(sb);
        return sb.ToString();
    }
}

File: examples/eApi v1.0/java/eAPI-v10-hu-connector/src/main/java/cz/monetplus/mips/eapi/v10hu/connector/entity/responses/OneclickEchoResponse.java
package cz.monetplus.mips.eapi.v10hu.connector.entity.responses;

import cz.monetplus.mips.eapi.v10hu.connector.entity.SignBase;
import cz.monetplus.mips.eapi.v10hu.connector.entity.ext.Extension;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

import java.util.List;

@EqualsAndHashCode(callSuper = false)
@Data
@AllArgsConstructor
@NoArgsConstructor
public class OneclickEchoResponse extends SignBase {

	private static final long serialVersionUID = -3825192932302805075L;
	
	private String origPayId;
	private int resultCode;
	private String resultMessage;
	private List<Extension> extensions;


	@Override
	public String toSign() {
		StringBuilder sb = new StringBuilder(); 
		add(sb, origPayId);
		add(sb, dttm);
		add(sb, resultCode);
		add(sb, resultMessage);
		deleteLast(sb);
		return sb.toString();
	}

}
